core/src/main/java/com/linecorp/armeria/client/cookie/DefaultCookieJar.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.cookie;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.CookieManager;
+import java.net.CookiePolicy;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableMap;
+
+import com.linecorp.armeria.common.Cookie;
+import com.linecorp.armeria.common.Cookies;
+
+/**
+ * A default in-memory {@link CookieJar} implementation that delegates to JDK's {@link CookieManager}.
+ * @see CookieManager
+ */
+final class DefaultCookieJar implements CookieJar {
+
+    private static final Map<String, List<String>> EMPTY_MAP = ImmutableMap.of();
+    private final CookieManager cookieManager;
+
+    DefaultCookieJar() {
+        cookieManager = new CookieManager(null, CookiePolicy.ACCEPT_ALL);
+    }
+
+    @Override
+    public Cookies get(URI uri) {
+        try {
+            final List<String> cookieList = cookieManager.get(uri, EMPTY_MAP).get("Cookie");
+            if (cookieList == null || cookieList.isEmpty()) {
+                return Cookies.of();
+            }
+            final List<Cookie> cookies = new ArrayList<>();
+            cookieList.forEach(c -> cookies.addAll(Cookie.fromCookieHeader(c)));
+            return Cookies.of(cookies);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @Override
+    public void set(URI uri, Cookies cookies) {
+        try {
+            cookieManager.put(uri, ImmutableMap.of("Set-Cookie", Cookie.toSetCookieHeaders(cookies)));
```

Comment:
I think this is very inefficient to convert the `Cookie` to String and the String to Cookie and to String again when in `get(...)`.
How about just implement our own `cookieManager`? I think it's not that hard to implement if we don't consider set-cookie2 header which is deprecated. (Armeria doesn't support it yet) Also, I think we can add the feature later if there's a demand. /cc @trustin, @ikhoon

Reply:
> How about just implement our own cookieManager? 

I prefer the Armeria optiomized version. We also should drop old cookies exceeding `max-age`.

Reply:
Thanks for the reviews 🙇‍♂️
Love the idea, I'll see to the implementation asap.

Reply:
@minwoox, @ikhoon, I've implemented a `CookieJar` that stores Armeria's `Cookie`. Please take a look 🙏